web/client/components/widgets/enhancers/tools/exportableWidget.js

```diff
@@ -19,12 +19,14 @@ module.exports = () =>
                 glyphClassName: "exportCSV",
                 target: "menu",
                 textId: "widgets.widget.menu.downloadData",
+                visible: data && data.length,
```

Comment:
Hi, with this you have found solution that avoids the problem which can be good, but have you investigated the cause?

can you explain why hiding is better than disabling?

Reply:
Because I discovered the problem was in widget epic ` exportWidgetImage` https://github.com/geosolutions-it/MapStore2/blob/c6e014a7c071c977e20b1385a25ad26de8256c73/web/client/epics/widgets.js#L144 
If you look at the epic code, you can notice that the epic logic is expecting the chart to be exported to be available in the document and it works well when that condition is fulfilled. One soln was to modify that logic to include the condition when the chart is not available in the dom or leave as it is and prevent it to be called when the chart is not available. I thought the latter was more appropriate since it solve the problem from the root. Why should someone export empty data/image? There is no such use case.

Reply:
i see, maybe we could also add a guard before start exporting to avoid the error

one other aspect is to leave the buttons visible but disabled and for this point i would ask confirmation of @tdipisa

Reply:
@MV88 sorry, I need to confess that I didn't understand your qn in the first place regarding why hiding and not disabling. You can ignore my last reply because it was answering something else. Well, back to your original question regarding hiding or disabling, I think it's a matter of preference and both ways works although I see more value on disabling only since it will give user knowledge that there is a possibility of exporting but it's only valid when there is data.

Reply:
@kasongoyo disabling is better than hiding the button but maybe a tooltip should be shown to inform the user why the button is disabled.

Reply:
@tdipisa i think that a disabled button has no possibility to show the tooltip

Reply:
ok